lib/oneview-sdk/image-streamer/resource/api300/golden_image.rb

```diff
@@ -10,13 +10,128 @@
 # language governing permissions and limitations under the License.
 
 require_relative 'resource'
+require 'net/http/post/multipart'
 
 module OneviewSDK
   module ImageStreamer
     module API300
       # Golden Image resource implementation for Image Streamer
       class GoldenImage < Resource
         BASE_URI = '/rest/golden-images'.freeze
+        READ_TIMEOUT = 300 # in seconds, 5 minutes
+        ACCEPTED_FORMATS = %w(.zip .ZIP).freeze # Supported upload extensions
+
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::ImageStreamer::Client] client The client object for the Image Streamer appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          super
+          # Default values:
+          @data['type'] ||= 'GoldenImage'
+        end
+
+        # Download the details of the golden image capture logs which has been archived based on the specific attribute ID.
+        # @param [String] file_path
+        # @return [True] When was saved successfully
+        def download_details_archive(file_path)
+          ensure_client && ensure_uri
+          resp = @client.rest_api(:get, "#{BASE_URI}/archive/#{@data['uri'].split('/').last}")
+          File.open(file_path, 'wb') { |file| file.write(resp.body) }
```

Comment:
@jsmartt , Jared, I made the change but I can only certify that it is correct when I test in real hardware. Let me know what you think of this solution?